src/Form/matching.js

```diff
@@ -393,10 +393,14 @@ class Matching {
      */
     * getElementStrings (el, form, opts = {}) {
         let {
-            matchableStrings = ['nameAttr', 'labelText', 'placeholderAttr', 'relatedText']
+            matchableStrings = ['id', 'nameAttr', 'labelText', 'placeholderAttr', 'relatedText']
```

Comment:
In the past we noticed that `id` is usually a rather noisy attribute, with tons of stuff inside. That can cause false positives. For example, I have seen `id` containing `login_address-street_zip`. That would match `address-street` instead of the zip code. There are also generated random ids for which it's easy to get `tr3zips`. That would again mismatch. Let's move `id` farther down the list:

```suggestion
            matchableStrings = ['nameAttr', 'labelText', 'placeholderAttr', 'id', 'relatedText']
```

I would also move the switch case, even though it doesn't matter. It's better if it follows the list above for consistency.

Reply:
In fact, wait a minute, let me test this in the real product. `id`s were causing lots of false positives in complex UIs like Asana or Litmus, where many `id`s contain stuff like `email` and were matching false positives.

Reply:
I was mostly just copying Firefox's ordering here and we run this after ddg and css selectors run.

> We can also add regression testing for this

Yeah we should codify these certainly. If you have sample pages I'm happy to add them.